src/framework/analyses.ml

```diff
@@ -417,6 +422,16 @@ module type GenericEqBoxSolver =
     val solve : (S.v -> S.d -> S.d -> S.d) -> (S.v*S.d) list -> S.v list -> S.d H.t
   end
 
+(* Like [GenericEqBoxSolver], but the solve method additionally returns an optional solver state for reuse in incremental runs *)
+module type GenericIncrEqBoxSolver =
+  functor (S:EqConstrSys) ->
+  functor (H:Hash.H with type key=S.v) ->
+  sig
+    (** The hash-map [solve box xs vs] is a local solution for interesting variables [vs],
+        reached from starting values [xs].  *)
+    val solve : (S.v -> S.d -> S.d -> S.d) -> (S.v*S.d) list -> S.v list -> S.d H.t * Obj.t option
+  end
+
```

Comment:
I now changed it so that the solvers other than TD3 return a dummy object. So no extra signatures anymore and no `Obj.t option`.

Reply:
> This extra signature and the `option` are a bit excessive. I think it'd be much simpler to just let every solver return a second `Obj.t` component (no optional). If it doesn't work incrementally and thus doesn't want to return any data for marshaling, it can just return `()`.

I think returning `None` is easier to understand than the now returned `Goblintutil.dummy_obj` in every other solver.
Maybe it makes sense to have some `IgnoreSolverState` functor or some other less ad-hoc solution.

Reply:
Not sure if this comment is somewhere here or elsewhere, but what we really need is for incremental and non-incremental solvers to have a different signature. One which doesn't make non-incremental solvers appear as if they are incremental.

This would also allow extracting the incremental pre- and post-processing out of TD3 into an incremental TD3, which calls the normal TD3 in the middle (wraps the non-incremental TD3) and doesn't have to contain any `FromSpec`-specific node identification hacks (which are _not_ generic!).
Especially since there's other incremental solver-specific logic for verify needed as well (issue #368).

Since all this is still quite open and prone to bigger refactoring anyway, then it's not worth doing much about it.